### ISSUE TITLE
Item Database: Include stacked item variants in database

### ISSUE THREAD

**Issue by osrsbox** [osrsbox/osrsbox-db#174]
A new bug was discovered in the item database where items that are stackable (e.g., `Coins`) that also have graphical icon differences do not have their stackable variants in the database. Take the example of `Coins` with the item ID of `995`. There is no additional item IDs in the database that cover the variations to the original `Coins` item... in different stacked amounts.

**Identifying stacked items**: It turns out that an `ItemDefinition` has two properties to identify an item with stacked variants: `countObj` and `countCo`. Examples for the `Coins` item is displayed below:

```
  "countCo": [
    2,
    3,
    4,
    5,
    25,
    100,
    250,
    1000,
    10000,
    0
  ],
  "countObj": [
    996,
    997,
    998,
    999,
    1000,
    1001,
    1002,
    1003,
    1004,
    0
  ],
```

The `countObj` property provides the variant item IDs, and the `countCo` property seems (at a best guess) to be a variable for the amount of a specific item to be represented by a different object.

**Including stacked items**: To include these items in the database, the following points need to be performed:

- The `generate_items_cache_data.py` script needs to be update to parse the `countObj` and `countCo` properties.
- Just adding the items will create a lot of duplicates, so a new property (e.g., `stacked` or `stacked_variant`) needs to be added and hold a boolean variable... or the item ID of the original item.
- Item icons need to be added... However, these icons seem to not be available using the current data dump method. More research here is needed.

**Comment by osrsbox** [comment]
About to push an update to resolve this issue. After putting in a lot of time to try and solve this, there are still a couple of issues/quirks to the solution. The following changes have been made:

- Update `generate_items_cache_data.py` to find `stacked` items
- Update `generate_items_cache_data.py` to label items with new `stacked` property
- Update `data/schemas` submodule to include new `stacked` property
- Update `osrsbox/items_api/item_properties` to include new `stacked` property
- Update `builders/items` to include new `stacked` property and handle missing item icons (use linked ID)
- This update added an additional 950 items

The remaining problems are:

- Sometimes an item links stacked variants with the same ID as other items (e.g., (p), (p+) and (p++) variant of arrows. In this case, the first occurrence is selected.
- No item icons are available from RuneLite HTTP API for stacked items, so the linked item ID icon is used (a new issue will be open to solve this problem). No other solution has been found to provide an up-to-date data source of item icons. The only solution that remains is to manually get all stacked icons.

**Comment by osrsbox** [comment]
Closed in a504a83fe81868fc73ce2766e14208d8dd68e2ca.
